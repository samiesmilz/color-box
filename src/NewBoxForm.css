input {
  padding: 5px;
  font-size: medium;
  margin: 0 5px;
}

button {
  padding: 5px 25px;
  font-size: medium;
}

.NewBoxForm {
  background-color: aquamarine;
  padding: 25px 10px;
  margin: 25px;
  border-radius: 5px;
  border: 1px black solid;
}
