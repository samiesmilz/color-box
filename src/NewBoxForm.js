import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import "./NewBoxForm.css";

const NewBoxForm = ({ addBox }) => {
  const INITIAL_STATE = {
    width: "",
    height: "",
    color: "",
  };
  const [data, SetData] = useState(INITIAL_STATE); // Manage state for form data

  const handleChange = (e) => {
    const { name, value } = e.target;
    SetData((formData) => ({ ...formData, [name]: value }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (data.width && data.height && data.color) {
      addBox({ ...data, id: uuid() });
      SetData(INITIAL_STATE);
    }
  };

  return (
    <div className="NewBoxForm">
      <form onSubmit={handleSubmit}>
        <label htmlFor="width">Set width:</label>
        <input
          type="text"
          id="width"
          name="width"
          placeholder="Enter width"
          value={data.width}
          onChange={handleChange}
        />
        <label htmlFor="height">Set height:</label>
        <input
          type="text"
          id="name"
          name="height"
          placeholder="Enter height"
          value={data.height}
          onChange={handleChange}
        />
        <label htmlFor="color">Set color:</label>
        <input
          type="text"
          id="color"
          name="color"
          placeholder="Enter color"
          value={data.color}
          onChange={handleChange}
        />
        <button type="submit" name="add-box">
          Add box
        </button>
      </form>
    </div>
  );
};
export default NewBoxForm;
