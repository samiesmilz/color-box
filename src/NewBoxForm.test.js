import { render, screen } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

it("renders withou errors", () => {
  render(<NewBoxForm />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});
