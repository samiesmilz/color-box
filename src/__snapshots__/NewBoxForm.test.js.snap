// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewBoxForm"
  >
    <form>
      <label
        for="width"
      >
        Set width:
      </label>
      <input
        id="width"
        name="width"
        placeholder="Enter width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Set height:
      </label>
      <input
        id="name"
        name="height"
        placeholder="Enter height"
        type="text"
        value=""
      />
      <label
        for="color"
      >
        Set color:
      </label>
      <input
        id="color"
        name="color"
        placeholder="Enter color"
        type="text"
        value=""
      />
      <button
        name="add-box"
        type="submit"
      >
        Add box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
